invenio_github/tasks.py

```diff
@@ -144,13 +144,15 @@ def process_release(release_id, verify_sender=False):
     except Exception as ex:
         error_handlers = current_github.release_error_handlers
         release.model.status = ReleaseStatus.FAILED
-
         for error_cls, handler in error_handlers + DEFAULT_ERROR_HANDLERS:
             if isinstance(ex, error_cls):
                 handler(release, ex)
                 current_app.logger.exception(
                     u'Error while processing GitHub release')
                 break
+        if error_cls is Exception:
+            db.session.commit()
+            process_release.retry(ex=ex)
```

Comment:
`.retry()` can [raise a `Retry` exception itself](https://docs.celeryproject.org/en/stable/userguide/tasks.html#retrying). Not sure what this means at this point (i.e. if the `finally` block runs)